Suspend Windows when Linux is suspending

oberien opened the pull request:
Fix #26

oberien committed:
Suspend Windows when Linux is suspending

Fix #26

main-- commented on misc/windows-gaming.sh:
```diff
@@ -0,0 +1,24 @@
+#!/bin/sh
+
+# This needs to be located in /usr/lib/systemd/system-sleep/
```
Quoting `systemd-suspend(8)`:

>Note that scripts or binaries dropped in /usr/lib/systemd/system-sleep/ are intended for local use only and should be considered hacks. If applications want to be notified of system suspend/hibernation and resume, there are much nicer interfaces available.

As I explained in the issue, a sleep inhibitor is pretty definitely what we want here.

main-- commented on guest-agent/VfioService/VfioService/ClientManager.cs:
```diff
@@ -47,6 +48,9 @@ public ClientManager(MainForm mainForm)
                         StuckKeyFix.ReleaseModifiers();
                         break;
+                    case CommandIn.Suspend:
+                        Application.SetSuspendState(PowerState.Suspend, true, true);
```
Why are these flags correct? Isn't `false, false` a better choice in our case?

main-- commented on misc/windows-gaming.sh:
```diff
@@ -0,0 +1,24 @@
+  fi
+  # Currently on arch netcat has a bug where it does not end the connection.
+  # Therefore we use a timeout of 3 seconds.
+  echo -ne "\x05" | $nc -w 3 -U /run/user/1000/windows-gaming-driver/control.sock
```
The solution for this is the `-N` flag as the server will obviously keep waiting for additional commands until the client closes the connection. (I believe the default behavior of arch's openbsd-netcat changed in a recent update so we now have to specify it but previously didn't.)

main-- commented on misc/windows-gaming.sh:
```diff
@@ -0,0 +1,24 @@
+  echo -ne "\x05" | $nc -w 3 -U /run/user/1000/windows-gaming-driver/control.sock
+  # We don't know how long windows needs to suspend, so just wait 8 secnods for now,
+  # which is "good enough for my case".
+  # TODO: Fix this once #67 is fixed
```
Again, a sleep inhibitor does not have this issue as it it runs inside the driver and can simply react to the qapi event.

main-- reviewed (changes requested):
I really think we need a sleep inhibitor.

Rest is looking great though.

oberien committed:
Deserialize QMP responses

main-- committed:
Implement a dbus-based systemd sleep inhibitor

main-- committed:
Explain to travis what a dbus is and where it can get one

main-- committed:
Merge pull request #78 from main--/games-dean

Implement a dbus-based systemd sleep inhibitor

oberien committed:
Make Controller::suspend return a future

It resolves when qemu is fully suspended.
Use it to suspend the guest when the host is suspended with dbus.

main-- committed:
Add the logind config to increase inhibitor timeout
